spy_integration_test.go

```diff
@@ -4,36 +4,112 @@ package spy_test
 
 import (
 	"bytes"
+	"fmt"
 	"spy"
 	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
 )
 
 func TestExecuteIntegration(t *testing.T) {
-	var buff bytes.Buffer
-	want := "echo Hello\nHello\n"
-	err := spy.Execute(&buff, "echo Hello")
+	commandLine := "echo Hello, World!"
+	wantRecorder := fmt.Sprintf("%s\nHello, World!\n", commandLine)
+	wantTerminal := "Hello, World!\n"
 
+	s, err := spy.NewSession("/tmp/test-file.log")
 	if err != nil {
-		t.Error(err)
+		t.Fatal(err)
 	}
 
-	got := buff.String()
-	if want != got {
-		t.Errorf("Want: %q, got: %q", want, got)
+	var logFile, terminal bytes.Buffer
+	s.Recorder = &logFile
+	s.Terminal = &terminal
+
+	err = s.Execute(commandLine)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	gotRecorder := logFile.String()
+	gotTerminal := terminal.String()
+
+	if !cmp.Equal(wantRecorder, gotRecorder) {
+		t.Error(cmp.Diff(wantRecorder, gotRecorder))
+	}
+
+	if !cmp.Equal(wantTerminal, gotTerminal) {
+		t.Error(cmp.Diff(wantTerminal, gotTerminal))
 	}
 }
 
 func TestExecuteIntegrationInvalid(t *testing.T) {
-	var buff bytes.Buffer
-	want := "echo Hello'\nError parsing command\n"
-	err := spy.Execute(&buff, "echo Hello'")
+	var logFile, terminal bytes.Buffer
+
+	want := "Error parsing command\n"
+	s, err := spy.NewSession("/tmp/test-file.log")
 
 	if err != nil {
 		t.Error(err)
 	}
+	s.Terminal = &terminal
+	s.Recorder = &logFile
+	err = s.Execute("echo Hello'")
 
-	got := buff.String()
+	if err != nil {
+		t.Error(err)
+	}
+
+	got := terminal.String()
 	if want != got {
 		t.Errorf("Want: %q, got: %q", want, got)
 	}
 }
+
+func TestWithTimestamp(t *testing.T) {
```

Comment:
This isn't an integration test, is it?

Reply:
Er..._sorta_. Each new session creates a file and testing this creates a temporary file.